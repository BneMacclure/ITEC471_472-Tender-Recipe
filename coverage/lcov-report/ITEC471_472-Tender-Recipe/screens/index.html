
<!doctype html>
<html lang="en">

<head>
    <title>Code coverage report for ITEC471_472-Tender-Recipe/screens</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../prettify.css" />
    <link rel="stylesheet" href="../../base.css" />
    <link rel="shortcut icon" type="image/x-icon" href="../../favicon.png" />
    <meta name="viewport" content="width=device-width, initial-scale=1" />
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../sort-arrow-sprite.png);
        }
    </style>
</head>
    
<body>
<div class='wrapper'>
    <div class='pad1'>
        <h1><a href="../../index.html">All files</a> ITEC471_472-Tender-Recipe/screens</h1>
        <div class='clearfix'>
            
            <div class='fl pad1y space-right2'>
                <span class="strong">26.45% </span>
                <span class="quiet">Statements</span>
                <span class='fraction'>114/431</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">3.66% </span>
                <span class="quiet">Branches</span>
                <span class='fraction'>3/82</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">9.15% </span>
                <span class="quiet">Functions</span>
                <span class='fraction'>15/164</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">26.51% </span>
                <span class="quiet">Lines</span>
                <span class='fraction'>114/430</span>
            </div>
        
            
        </div>
        <p class="quiet">
            Press <em>n</em> or <em>j</em> to go to the next uncovered block, <em>b</em>, <em>p</em> or <em>k</em> for the previous block.
        </p>
    </div>
    <div class='status-line low'></div>
    <div class="pad1">
<table class="coverage-summary">
<thead>
<tr>
   <th data-col="file" data-fmt="html" data-html="true" class="file">File</th>
   <th data-col="pic" data-type="number" data-fmt="html" data-html="true" class="pic"></th>
   <th data-col="statements" data-type="number" data-fmt="pct" class="pct">Statements</th>
   <th data-col="statements_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="branches" data-type="number" data-fmt="pct" class="pct">Branches</th>
   <th data-col="branches_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="functions" data-type="number" data-fmt="pct" class="pct">Functions</th>
   <th data-col="functions_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="lines" data-type="number" data-fmt="pct" class="pct">Lines</th>
   <th data-col="lines_raw" data-type="number" data-fmt="html" class="abs"></th>
</tr>
</thead>
<tbody><tr>
	<td class="file low" data-value="CreateRecipeScreen.js"><a href="CreateRecipeScreen.js.html">CreateRecipeScreen.js</a></td>
	<td data-value="31.75" class="pic low">
	<div class="chart"><div class="cover-fill" style="width: 31%"></div><div class="cover-empty" style="width: 69%"></div></div>
	</td>
	<td data-value="31.75" class="pct low">31.75%</td>
	<td data-value="63" class="abs low">20/63</td>
	<td data-value="7.14" class="pct low">7.14%</td>
	<td data-value="14" class="abs low">1/14</td>
	<td data-value="4.35" class="pct low">4.35%</td>
	<td data-value="23" class="abs low">1/23</td>
	<td data-value="31.75" class="pct low">31.75%</td>
	<td data-value="63" class="abs low">20/63</td>
	</tr>

<tr>
	<td class="file low" data-value="LoginScreen.js"><a href="LoginScreen.js.html">LoginScreen.js</a></td>
	<td data-value="32.35" class="pic low">
	<div class="chart"><div class="cover-fill" style="width: 32%"></div><div class="cover-empty" style="width: 68%"></div></div>
	</td>
	<td data-value="32.35" class="pct low">32.35%</td>
	<td data-value="34" class="abs low">11/34</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="4" class="abs low">0/4</td>
	<td data-value="5.26" class="pct low">5.26%</td>
	<td data-value="19" class="abs low">1/19</td>
	<td data-value="32.35" class="pct low">32.35%</td>
	<td data-value="34" class="abs low">11/34</td>
	</tr>

<tr>
	<td class="file low" data-value="MainScreen.js"><a href="MainScreen.js.html">MainScreen.js</a></td>
	<td data-value="27.08" class="pic low">
	<div class="chart"><div class="cover-fill" style="width: 27%"></div><div class="cover-empty" style="width: 73%"></div></div>
	</td>
	<td data-value="27.08" class="pct low">27.08%</td>
	<td data-value="96" class="abs low">26/96</td>
	<td data-value="6.25" class="pct low">6.25%</td>
	<td data-value="32" class="abs low">2/32</td>
	<td data-value="16.22" class="pct low">16.22%</td>
	<td data-value="37" class="abs low">6/37</td>
	<td data-value="27.37" class="pct low">27.37%</td>
	<td data-value="95" class="abs low">26/95</td>
	</tr>

<tr>
	<td class="file low" data-value="MealPlannerScreen.js"><a href="MealPlannerScreen.js.html">MealPlannerScreen.js</a></td>
	<td data-value="35.9" class="pic low">
	<div class="chart"><div class="cover-fill" style="width: 35%"></div><div class="cover-empty" style="width: 65%"></div></div>
	</td>
	<td data-value="35.9" class="pct low">35.9%</td>
	<td data-value="39" class="abs low">14/39</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="2" class="abs low">0/2</td>
	<td data-value="12.5" class="pct low">12.5%</td>
	<td data-value="16" class="abs low">2/16</td>
	<td data-value="35.9" class="pct low">35.9%</td>
	<td data-value="39" class="abs low">14/39</td>
	</tr>

<tr>
	<td class="file low" data-value="MyProfile.js"><a href="MyProfile.js.html">MyProfile.js</a></td>
	<td data-value="48" class="pic low">
	<div class="chart"><div class="cover-fill" style="width: 48%"></div><div class="cover-empty" style="width: 52%"></div></div>
	</td>
	<td data-value="48" class="pct low">48%</td>
	<td data-value="25" class="abs low">12/25</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="12.5" class="pct low">12.5%</td>
	<td data-value="8" class="abs low">1/8</td>
	<td data-value="48" class="pct low">48%</td>
	<td data-value="25" class="abs low">12/25</td>
	</tr>

<tr>
	<td class="file low" data-value="MyRecipes.js"><a href="MyRecipes.js.html">MyRecipes.js</a></td>
	<td data-value="12.36" class="pic low">
	<div class="chart"><div class="cover-fill" style="width: 12%"></div><div class="cover-empty" style="width: 88%"></div></div>
	</td>
	<td data-value="12.36" class="pct low">12.36%</td>
	<td data-value="89" class="abs low">11/89</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="28" class="abs low">0/28</td>
	<td data-value="15" class="pct low">15%</td>
	<td data-value="20" class="abs low">3/20</td>
	<td data-value="12.36" class="pct low">12.36%</td>
	<td data-value="89" class="abs low">11/89</td>
	</tr>

<tr>
	<td class="file medium" data-value="RegisterScreen.js"><a href="RegisterScreen.js.html">RegisterScreen.js</a></td>
	<td data-value="50" class="pic medium">
	<div class="chart"><div class="cover-fill" style="width: 50%"></div><div class="cover-empty" style="width: 50%"></div></div>
	</td>
	<td data-value="50" class="pct medium">50%</td>
	<td data-value="18" class="abs medium">9/18</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="2" class="abs low">0/2</td>
	<td data-value="12.5" class="pct low">12.5%</td>
	<td data-value="8" class="abs low">1/8</td>
	<td data-value="50" class="pct medium">50%</td>
	<td data-value="18" class="abs medium">9/18</td>
	</tr>

<tr>
	<td class="file low" data-value="Registration02.js"><a href="Registration02.js.html">Registration02.js</a></td>
	<td data-value="18.18" class="pic low">
	<div class="chart"><div class="cover-fill" style="width: 18%"></div><div class="cover-empty" style="width: 82%"></div></div>
	</td>
	<td data-value="18.18" class="pct low">18.18%</td>
	<td data-value="11" class="abs low">2/11</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="6" class="abs low">0/6</td>
	<td data-value="18.18" class="pct low">18.18%</td>
	<td data-value="11" class="abs low">2/11</td>
	</tr>

<tr>
	<td class="file low" data-value="Registration03.js"><a href="Registration03.js.html">Registration03.js</a></td>
	<td data-value="27.27" class="pic low">
	<div class="chart"><div class="cover-fill" style="width: 27%"></div><div class="cover-empty" style="width: 73%"></div></div>
	</td>
	<td data-value="27.27" class="pct low">27.27%</td>
	<td data-value="11" class="abs low">3/11</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="5" class="abs low">0/5</td>
	<td data-value="27.27" class="pct low">27.27%</td>
	<td data-value="11" class="abs low">3/11</td>
	</tr>

<tr>
	<td class="file low" data-value="Registration04.js"><a href="Registration04.js.html">Registration04.js</a></td>
	<td data-value="25" class="pic low">
	<div class="chart"><div class="cover-fill" style="width: 25%"></div><div class="cover-empty" style="width: 75%"></div></div>
	</td>
	<td data-value="25" class="pct low">25%</td>
	<td data-value="12" class="abs low">3/12</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="6" class="abs low">0/6</td>
	<td data-value="25" class="pct low">25%</td>
	<td data-value="12" class="abs low">3/12</td>
	</tr>

<tr>
	<td class="file low" data-value="Registration05.js"><a href="Registration05.js.html">Registration05.js</a></td>
	<td data-value="9.09" class="pic low">
	<div class="chart"><div class="cover-fill" style="width: 9%"></div><div class="cover-empty" style="width: 91%"></div></div>
	</td>
	<td data-value="9.09" class="pct low">9.09%</td>
	<td data-value="33" class="abs low">3/33</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="16" class="abs low">0/16</td>
	<td data-value="9.09" class="pct low">9.09%</td>
	<td data-value="33" class="abs low">3/33</td>
	</tr>

</tbody>
</table>
</div>
                <div class='push'></div><!-- for sticky footer -->
            </div><!-- /wrapper -->
            <div class='footer quiet pad2 space-top1 center small'>
                Code coverage generated by
                <a href="https://istanbul.js.org/" target="_blank">istanbul</a>
                at Wed Mar 17 2021 23:16:30 GMT-0400 (Eastern Daylight Time)
            </div>
        </div>
        <script src="../../prettify.js"></script>
        <script>
            window.onload = function () {
                prettyPrint();
            };
        </script>
        <script src="../../sorter.js"></script>
        <script src="../../block-navigation.js"></script>
    </body>
</html>
    