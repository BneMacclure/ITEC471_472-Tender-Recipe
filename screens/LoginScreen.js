import React, { Component , useState} from "react";
import {
    StyleSheet,
    View,
    Image,
    ImageBackground,
    Text,
    TextInput, 
    TouchableOpacity,
    ActivityIndicator,
    Button,
    Alert,
    StatusBar,
    Dimensions
} from "react-native";
import IoniconsIcon from "react-native-vector-icons/Ionicons";
import EntypoIcon from "react-native-vector-icons/Entypo";
import {firebaseApp} from '../config/DatabaseConfig';
import styles from '../styles/LoginStyles.js';

const SCREEN_HEIGHT = Dimensions.get('window').height
const SCREEN_WIDTH = Dimensions.get('window').width

const LoginScreen = (props) => {
    //
    const [loading, setLoading] = useState(false);
    const startLoading = () => {
        setLoading(true);
        setTimeout(() => {
            setLoading(false);
        }, 3000);
    };
    const [email, setEmail] = useState(""); // email for login
    const [password, setPassword] = useState(""); // password for login
    // Function for login. Uses firebase authentication
    const loginFunc = () => {
      //activates the activity indicator's animation
      startLoading();
      // check if valid login
      firebaseApp
      .auth()
      .signInWithEmailAndPassword(email, password)
      .then(() => props.navigation.navigate('Main Screen'))
      .catch(error => Alert.alert(
          "Login",
          "Invalid login information",
          [
            { text: "OK", onPress: () => console.log("OK Pressed") }
          ],
          { cancelable: false }
        )
      )   
    };

    // Function for login. Uses firebase authentication
    const debugLoginFunc = () => {
        //activates the activity indicator's animation
        startLoading();
        // check if valid login
        firebaseApp
        .auth()
        .signInWithEmailAndPassword("test@radford.edu", "password")
        .then(() => props.navigation.navigate('Main Screen'))
        .catch(error => Alert.alert(
            "Login",
            "Invalid login information",
            [
              { text: "OK", onPress: () => console.log("OK Pressed") }
            ],
            { cancelable: false }
          )
        )   
      };

    // Function for registering an account. Uses firbase's authentication API to signup
    const registerFunc = () => {
        if(email === '' && password === '') {
            Alert.alert('Enter details to signup!')
          } 
        else {
            firebaseApp
            .auth()
            .createUserWithEmailAndPassword(email, password)
            .then(() => Alert.alert(
                "Register",
                "You're registered! Enjoy!",
                [
                    {text: "OK", onPress: () => console.log("Register OK pressed") }
                ],
                { cancelable: false }
            ))
            .catch(error => Alert.alert(
                "Register",
                "Invalid information",
                [
                    {text: "OK", onPress: () => console.log("OK pressed") }
                ],
                { cancelable: false }
            ))
        }
    };

    return (
        <View style={styles.container}>
            <StatusBar
                backgroundColor="#E35614"
                />
            <ImageBackground
                source={require("../assets/images/login_bg.png")}
                resizeMode="cover"
                style={styles.image}
                imageStyle={styles.image_imageStyle}
            >
                <Image
                    source={require("../assets/images/logo.png")}
                    resizeMode="contain"
                    style={styles.logo}
                ></Image>
                <Text style={styles.tenderRecipes}>Tender Recipes</Text>


                <View style={styles.usernameGroup}>
                    <IoniconsIcon name="md-person" style={styles.icon}></IoniconsIcon>
                    <TextInput
                        testID='emailInput'
                        placeholder="Email"
                        placeholderTextColor="rgba(230, 230, 230,1)"
                        inlineImagePadding={0}
                        selectionColor="rgba(52,250,215,1)"
                        style={styles.textInput}
                        onChangeText={(email) => setEmail(email)}
                    ></TextInput>
                </View>
                <View style={styles.passwordGroup}>
                    <EntypoIcon name="lock" style={styles.icon2}></EntypoIcon>
                    <TextInput
                        testID='passwordInput'
                        placeholder="Password"
                        placeholderTextColor="rgba(230, 230, 230,1)"
                        selectionColor="rgba(52,250,215,1)"
                        secureTextEntry={true}
                        style={styles.textInput2}
                        onChangeText={(password) => setPassword(password)}
                    ></TextInput>
                </View>


                <View style={styles.loginBtnRow}>
                    <TouchableOpacity
                        onPress={ loginFunc } //insert navigation
                        style={styles.loginBtn}
                        testID='loginButton'
                    >
                        <Text style={styles.login}>Login</Text>
                    </TouchableOpacity>
                    <TouchableOpacity
                        onPress={() => props.navigation.navigate('RegisterScreen')}
                        testID='registerButton'
                        //onPress={registerFunc} //insert navigation
                        style={styles.signupBtn}
                    >
                        <Text style={styles.signUp}>Sign up</Text>
                    </TouchableOpacity>
                    {/*DEBUG BUTTON*/}
                    <TouchableOpacity
                        onPress={ debugLoginFunc }
                        //onPress={registerFunc} //insert navigation
                        style={styles.signupBtn}
                    >
                        <Text style={styles.signUp}>Debug Login</Text>
                    </TouchableOpacity>
                </View>

                <ActivityIndicator
                    animating={loading}
                    color="rgba(254,242,94,1)"
                    size="large"
                    style={styles.activityIndicator}
                ></ActivityIndicator>

            </ImageBackground>
        </View>
    );
}

export default LoginScreen;