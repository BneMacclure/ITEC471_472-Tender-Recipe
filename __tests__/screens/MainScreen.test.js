import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import MainScreenInfo from '../../screens/MainScreen';
import { renderRecipes } from '../../screens/MainScreen';
// in a test setup file, or your test itself
const FRAME_TIME = 10

global.requestAnimationFrame = cb => {
  setTimeout(cb, FRAME_TIME)
}

describe('<MainScreen />', () => {
    it('main screen snapshot', () => {
        const tree = renderer.create(<MainScreenInfo />).toJSON();
        expect(tree).toMatchSnapshot();
    });
	it('main screen snapshot', () => {
		let main = new MainScreenInfo();
		
        expect(main.renderRecipes()).tobenull();
    });
	it('first registertion page renders',()=>{
		const wrapper = render(<MainScreenInfo />);
		expect(wrapper).toBeTruthy();
	});
})