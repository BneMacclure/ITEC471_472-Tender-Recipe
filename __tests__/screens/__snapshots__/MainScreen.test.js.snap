// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MainScreen /> main screen snapshot 1`] = `
<View
  accessibilityIgnoresInvertColors={true}
  style={
    Object {
      "alignSelf": "center",
      "height": "100%",
      "width": "100%",
    }
  }
>
  <Image
    resizeMode="cover"
    source={1}
    style={
      Array [
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        },
        Object {
          "height": "100%",
          "width": "100%",
        },
        Object {},
      ]
    }
  />
  <View
    style={
      Object {
        "flex": 1,
      }
    }
    testID="testLocation1"
  >
    <Modal
      animationType="slide"
      deviceHeight={null}
      deviceWidth={null}
      hardwareAccelerated={false}
      hideModalContentWhileAnimating={false}
      onBackdropPress={[Function]}
      onModalHide={[Function]}
      onModalWillHide={[Function]}
      onModalWillShow={[Function]}
      onRequestClose={[Function]}
      scrollHorizontal={false}
      scrollOffset={0}
      scrollOffsetMax={0}
      scrollTo={null}
      supportedOrientations={
        Array [
          "portrait",
          "landscape",
        ]
      }
      swipeThreshold={100}
      transparent={true}
      visible={false}
    >
      <View
        accessible={true}
        focusable={true}
        forwardedRef={[Function]}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "backgroundColor": "#000",
            "bottom": 0,
            "height": 1334,
            "left": 0,
            "opacity": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
            "width": 750,
          }
        }
      />
      <View
        animationType="slide"
        deviceHeight={null}
        deviceWidth={null}
        forwardedRef={[Function]}
        hideModalContentWhileAnimating={false}
        onBackdropPress={[Function]}
        onModalHide={[Function]}
        onModalWillHide={[Function]}
        onModalWillShow={[Function]}
        onRequestClose={[Function]}
        pointerEvents="box-none"
        scrollHorizontal={false}
        scrollOffset={0}
        scrollOffsetMax={0}
        scrollTo={null}
        style={
          Object {
            "flex": 1,
            "justifyContent": "center",
            "margin": 37.5,
            "transform": Array [
              Object {
                "translateY": 0,
              },
            ],
          }
        }
        supportedOrientations={
          Array [
            "portrait",
            "landscape",
          ]
        }
        swipeThreshold={100}
        transparent={true}
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flex": 1,
              "justifyContent": "center",
            }
          }
        >
          <View
            style={
              Object {
                "backgroundColor": "#fff",
                "borderColor": "#000",
                "borderRadius": 1,
                "borderStyle": "dashed",
                "borderWidth": 2,
                "height": "50%",
                "width": "100%",
              }
            }
          >
            <RCTScrollView
              style={
                Object {
                  "paddingTop": 30,
                }
              }
            >
              <View>
                <Text
                  style={
                    Object {
                      "fontSize": 22,
                      "fontWeight": "300",
                      "marginBottom": 20,
                      "textAlign": "center",
                    }
                  }
                >
                  Berry Tart
                </Text>
                <View>
                  <View
                    style={Object {}}
                  >
                    <View
                      accessible={true}
                      focusable={true}
                      onClick={[Function]}
                      onResponderGrant={[Function]}
                      onResponderMove={[Function]}
                      onResponderRelease={[Function]}
                      onResponderTerminate={[Function]}
                      onResponderTerminationRequest={[Function]}
                      onStartShouldSetResponder={[Function]}
                      style={
                        Object {
                          "opacity": 1,
                        }
                      }
                    >
                      <View
                        forwardedRef={[Function]}
                        style={
                          Object {
                            "backgroundColor": "#FD8014",
                            "marginBottom": 5,
                            "padding": 20,
                          }
                        }
                      >
                        <Text
                          style={
                            Object {
                              "color": "white",
                              "fontSize": 20,
                              "fontWeight": "700",
                              "textAlign": "center",
                            }
                          }
                        >
                          Allergens
                        </Text>
                      </View>
                    </View>
                    <View
                      pointerEvents="none"
                      style={
                        Object {
                          "height": 0,
                          "overflow": "hidden",
                        }
                      }
                    >
                      <View
                        forwardedRef={[Function]}
                        onLayout={[Function]}
                        style={Object {}}
                      >
                        <View
                          forwardedRef={[Function]}
                          style={
                            Object {
                              "backgroundColor": "rgba(245,252,255,1)",
                              "marginBottom": 5,
                              "padding": 20,
                            }
                          }
                        >
                          <Text
                            forwardedRef={[Function]}
                            style={Object {}}
                          >
                            Dairy, Gluten
                          </Text>
                        </View>
                      </View>
                    </View>
                  </View>
                  <View
                    style={Object {}}
                  >
                    <View
                      accessible={true}
                      focusable={true}
                      onClick={[Function]}
                      onResponderGrant={[Function]}
                      onResponderMove={[Function]}
                      onResponderRelease={[Function]}
                      onResponderTerminate={[Function]}
                      onResponderTerminationRequest={[Function]}
                      onStartShouldSetResponder={[Function]}
                      style={
                        Object {
                          "opacity": 1,
                        }
                      }
                    >
                      <View
                        forwardedRef={[Function]}
                        style={
                          Object {
                            "backgroundColor": "#FD8014",
                            "marginBottom": 5,
                            "padding": 20,
                          }
                        }
                      >
                        <Text
                          style={
                            Object {
                              "color": "white",
                              "fontSize": 20,
                              "fontWeight": "700",
                              "textAlign": "center",
                            }
                          }
                        >
                          Ingredients
                        </Text>
                      </View>
                    </View>
                    <View
                      pointerEvents="none"
                      style={
                        Object {
                          "height": 0,
                          "overflow": "hidden",
                        }
                      }
                    >
                      <View
                        forwardedRef={[Function]}
                        onLayout={[Function]}
                        style={Object {}}
                      >
                        <View
                          forwardedRef={[Function]}
                          style={
                            Object {
                              "backgroundColor": "rgba(245,252,255,1)",
                              "marginBottom": 5,
                              "padding": 20,
                            }
                          }
                        >
                          <Text
                            forwardedRef={[Function]}
                            style={Object {}}
                          >
                            2 cups of berries, 2 cups of butter, 1.5 cups of AP flour, 1 egg, 1 cup sugar
                          </Text>
                        </View>
                      </View>
                    </View>
                  </View>
                  <View
                    style={Object {}}
                  >
                    <View
                      accessible={true}
                      focusable={true}
                      onClick={[Function]}
                      onResponderGrant={[Function]}
                      onResponderMove={[Function]}
                      onResponderRelease={[Function]}
                      onResponderTerminate={[Function]}
                      onResponderTerminationRequest={[Function]}
                      onStartShouldSetResponder={[Function]}
                      style={
                        Object {
                          "opacity": 1,
                        }
                      }
                    >
                      <View
                        forwardedRef={[Function]}
                        style={
                          Object {
                            "backgroundColor": "#FD8014",
                            "marginBottom": 5,
                            "padding": 20,
                          }
                        }
                      >
                        <Text
                          style={
                            Object {
                              "color": "white",
                              "fontSize": 20,
                              "fontWeight": "700",
                              "textAlign": "center",
                            }
                          }
                        >
                          Instructions
                        </Text>
                      </View>
                    </View>
                    <View
                      pointerEvents="none"
                      style={
                        Object {
                          "height": 0,
                          "overflow": "hidden",
                        }
                      }
                    >
                      <View
                        forwardedRef={[Function]}
                        onLayout={[Function]}
                        style={Object {}}
                      >
                        <View
                          forwardedRef={[Function]}
                          style={
                            Object {
                              "backgroundColor": "rgba(245,252,255,1)",
                              "marginBottom": 5,
                              "padding": 20,
                            }
                          }
                        >
                          <Text
                            forwardedRef={[Function]}
                            style={Object {}}
                          >
                            Mix the flour and butter to form crust. Chill. Cook the berries. Add to crust. Bake immediately.
                          </Text>
                        </View>
                      </View>
                    </View>
                  </View>
                </View>
                <View
                  style={
                    Object {
                      "alignSelf": "center",
                      "height": 196,
                      "paddingTop": 30,
                      "width": "80%",
                    }
                  }
                >
                  <View
                    accessible={true}
                    focusable={true}
                    onClick={[Function]}
                    onResponderGrant={[Function]}
                    onResponderMove={[Function]}
                    onResponderRelease={[Function]}
                    onResponderTerminate={[Function]}
                    onResponderTerminationRequest={[Function]}
                    onStartShouldSetResponder={[Function]}
                    style={
                      Object {
                        "backgroundColor": "#e35514",
                        "borderColor": "#000",
                        "borderRadius": 10,
                        "borderWidth": 1,
                        "height": 56,
                        "justifyContent": "center",
                        "opacity": 1,
                      }
                    }
                  >
                    <Text
                      style={
                        Object {
                          "alignSelf": "center",
                          "color": "rgba(255,255,255,1)",
                          "fontWeight": "bold",
                        }
                      }
                    >
                      Exit View
                    </Text>
                  </View>
                </View>
              </View>
            </RCTScrollView>
          </View>
        </View>
      </View>
    </Modal>
    <View
      style={
        Object {
          "height": 10,
        }
      }
    />
    <View
      style={
        Object {
          "flex": 1,
        }
      }
      testID="recipeStackView"
    />
    <View
      style={
        Object {
          "height": 10,
        }
      }
    />
    <View
      style={
        Object {
          "alignSelf": "center",
          "flexDirection": "row",
          "height": 60,
          "marginTop": 65,
          "padding": 70,
        }
      }
    >
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "backgroundColor": "#f94723",
            "borderRadius": 100,
            "height": 60,
            "opacity": 1,
            "width": 64,
          }
        }
      >
        <Text />
      </View>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "backgroundColor": "#f94723",
            "borderRadius": 100,
            "height": 60,
            "marginLeft": 43,
            "opacity": 1,
            "width": 64,
          }
        }
      >
        <Text />
      </View>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "backgroundColor": "#f94723",
            "borderRadius": 100,
            "height": 60,
            "justifyContent": "center",
            "marginLeft": 39,
            "opacity": 1,
            "width": 64,
          }
        }
      >
        <Text />
      </View>
    </View>
  </View>
</View>
`;
